stdpopsim/slim_engine.py

```diff
@@ -900,7 +900,9 @@ def fix_time(event):
     growth_rates = np.empty(shape=(dd.num_populations, len(epochs)), dtype=float)
     for j, epoch in enumerate(epochs):
         for i, pop in enumerate(epoch.populations):
-            N[i, j] = int(pop.end_size)
+            # SLiM simulates a diploid population, so rescale population size
+            # depending on contig ploidy.
+            N[i, j] = int(pop.end_size * contig.ploidy / 2)
```

Comment:
I agree with this, although we might want to leave a note saying that this should be changed if/when we get SLiM to do haploids correctly (in an appropriate issue).